diff --git a/internal/ollama/ollama.go b/internal/ollama/ollama.go
index 949840c..b617094 100644
--- a/internal/ollama/ollama.go
+++ b/internal/ollama/ollama.go
@@ -164,6 +172,7 @@ func (s *Stream) Next() bool {
 		s.messages = append(s.messages, toProtoMessage(s.message))
 		s.request.Messages = append(s.request.Messages, s.message)
 		s.message = api.Message{}
+		return false
 	}
 	return true
 }